BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -4006,8 +4006,10 @@ void CheckAIAutofire()
                 Vector3 aimDirection = fireTransform.forward;
                 targetCosAngle = Vector3.Dot(aimDirection, targetRelPos.normalized);
                 var maxAutoFireCosAngle2 = targetAdjustedMaxCosAngle;
-
                 safeToFire = CheckForFriendlies(fireTransform);
+                if (BDArmorySettings.BULLET_WATER_DRAG && FlightGlobals.getAltitudeAtPos(fireTransforms[0].position) < 0 && !visualTargetVessel.IsUnderwater())
```

Comment:
Can `visualTargetVessel` be `null` here? If so, then it'll need a null check.

Reply:
if Guardmode was somehow disabled/vessel was destroyed while in the middle of executing AimAndFire() (so after  UpdateTargetvessel() but before CheckAIAutofire() ) So possibly, if abet incredibly improbably?

Reply:
`AimAndFire` runs in the `FashionablyLate` timing phase of `FixedUpdate`, so `guardMode` getting disabled or the vessel getting destroyed shouldn't be possible. So, this should be fine. We can always deal with it later if it does throw an NRE.